lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -103,9 +103,18 @@ def bulk(actions)
 
       return if actions.empty?
 
+      unsupported_action_indexes = []
+      index = 0
+
       bulk_actions = actions.collect do |action, args, source|
         args, source = update_action_builder(args, source) if action == 'update'
 
+        # do not filter out the action, we have a logic in common.rb -> submit() to match action index with response index
+        unless ElasticSearch::VALID_HTTP_ACTIONS.include?(action)
+          unsupported_action_indexes << index
+        end
+        index = index + 1
```

Comment:
somehow don't like this plain incremental approach but when I tried `actions.collect.with_index`, return data (`bulk_actions`) messed up. Any way to simplify?